tendrl/ceph_integration/objects/rbd/atoms/create/__init__.py

```diff
@@ -13,6 +17,54 @@ def __init__(self, *args, **kwargs):
         super(Create, self).__init__(*args, **kwargs)
 
     def run(self):
+        # Checking if a new  pool has to be created or not rbd creation.
+        if self.parameters['Rbd.create_pool'] is True and \
+                        self.parameters['Rbd.pool_id'] is None:
+            # Creating job for pool creation.
+            if ['Rbd.pool_parameters'] in self.parameters:
```

Comment:
In that case use pool parameters as first class "optional" inputs instead of "mandatory"

dont use the pool_parameters dict